src/core/providers/ogr/qgsogrproviderutils.cpp

```diff
@@ -2826,78 +2826,95 @@ GIntBig QgsOgrLayer::GetApproxFeatureCount()
   QString driverName = GDALGetDriverShortName( GDALGetDatasetDriver( ds->hDS ) );
   if ( driverName == QLatin1String( "GPKG" ) )
   {
-    CPLPushErrorHandler( CPLQuietErrorHandler );
-    OGRLayerH hSqlLayer = GDALDatasetExecuteSQL(
-                            ds->hDS, "SELECT 1 FROM gpkg_ogr_contents LIMIT 0", nullptr, nullptr );
-    CPLPopErrorHandler();
-    if ( hSqlLayer )
-    {
-      GDALDatasetReleaseResultSet( ds->hDS, hSqlLayer );
-      return OGR_L_GetFeatureCount( hLayer, TRUE );
-    }
+    GIntBig totalFeatureCount = getTotalFeatureCountfromMetaData();
```

Comment:
Can you open a separate PR with just this change? It'll be easier to discuss/review it in isolation from the other changes.

Reply:
-> I moved it to https://github.com/qgis/QGIS/pull/48099